app/src/main/java/net/skyscanner/backpack/demo/compose/DialogStory.kt

```diff
@@ -0,0 +1,167 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.demo.compose
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.padding
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import net.skyscanner.backpack.compose.button.BpkButton
+import net.skyscanner.backpack.compose.dialog.BpkAlertDialog
+import net.skyscanner.backpack.compose.dialog.BpkDestructiveDialog
+import net.skyscanner.backpack.compose.dialog.BpkSuccessDialog
+import net.skyscanner.backpack.compose.dialog.DialogButton
+import net.skyscanner.backpack.compose.icons.BpkIcons
+import net.skyscanner.backpack.compose.icons.lg.AlertAdd
+import net.skyscanner.backpack.compose.icons.lg.Tick
+import net.skyscanner.backpack.compose.icons.lg.Trash
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import net.skyscanner.backpack.demo.R
+
+@Composable
+fun DialogStory() {
+  Column(
+    modifier = Modifier.padding(BpkSpacing.Base),
+    verticalArrangement = Arrangement.spacedBy(BpkSpacing.Base),
+  ) {
+    var dialogShown by rememberSaveable { mutableStateOf<Dialog?>(null) }
+    Dialog.values().forEach {
+      BpkButton(it.buttonText) {
+        dialogShown = it
+      }
+    }
+
+    when (dialogShown) {
+      Dialog.SuccessOneButton -> SuccessOneButtonDialogExample { dialogShown = null }
```

Comment:
Can we rename it to showing dialogue type, and add an additional option for None? It'll make the code look cleaner

Reply:
that'll make the creation of them more difficult as the enum is also there for the button creation